src/app/data-model-provider/tests/BUILD.gn

```diff
@@ -30,3 +30,30 @@ chip_test_suite("tests") {
     "${chip_root}/src/lib/core:string-builder-adapters",
   ]
 }
+
+source_set("encode-decode") {
+  # Handles the conversions between Raw data (usually byte buffers backed by TLV encoders
+  # and `*IBs::Builder` classes) and data used by by Providers like AttributeValueEncoder
+  # or AttributeValueDecoder.
+  #
+  # Intended to make writing tests simpler and focusing less on encoding and more on
+  # "Assume you receive an int" or "Check that the encoded value was the string 'abc'"
+  # and so on
+  sources = [
+    "ReadTesting.cpp",
```

Comment:
nit: all test files are usually called TestXxxx.cpp

Reply:
In this particular case, these are support files, so a file `TestFoo` would include/use `ReadTesting` to test `DataModel::Provider::Read` code.

I think having a `Test` prefix would make these look like actual tests and they are not.